keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -254,12 +263,138 @@ tried this feature in Alpha, we would have time to fix issues.
+
+
 ## Graduation Criteria
 
 - Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default. 
 - Beta:  Enabled by default with default value of 1. 
 
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: MaxUnavailableStatefulSet
+  - Components depending on the feature gate: StatefulSet
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior remains the same.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled. Once disabled, all existing  StatefulSet will
+revert to the old behavior where rolling update will proceed one pod at a time.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+If the feature is enabled and then later disabled, we will revert the value of 
+maxUnavailable to the default value so that customers are not surprised.
+
+###### Are there any tests for feature enablement/disablement?
+yes, there are unit tests which make sure the field is correctly dropped 
+on feature enable and disabled
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+MaxUnavailable can fail to rollout if the new image being deployed is bad and causes the 
+containers to crash. If there are not enough instances, it can bring down many or 
+cause disruption if the maxUnavailable is set to a large number, larger than what 
+the service can tolerate. 
+If the StatefulSet is already in a failed state with more than maxUnavailable pods 
+failing, changing maxUnavailable will not help, and it will require first fixing the 
+failing pod before the system can recover.
+
+It can impact already running workloads if service owner accidentally sets maxUnavailable
+to a very high number.
+
+###### What specific metrics should inform a rollback?
+The customer should look at the behavior of the maxUnavailable and if the rollout
+is not respecting the maxUnavailable, the feature should be rolledback.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+No
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+If their StatefulSet rollingUpdate section has the field maxUnavailable specified with
+a value different than 1.
+The below command should show maxUnavailable value:
+```
+kubectl get statefulsets -o yaml | grep maxUnavailable
+```
+
+###### How can someone using this feature know that it is working for their instance?
+Ensure the number of replicas in your StatefulSet is more than 2. 
```

Comment:
This isn't really what we want. We don't ask how to test the behavior.

We're rather asking how to check if the feature is generally working.

[The question is not strictly required for Alpha - but basically I would expect some metric to be exposed from statefule-set controller that will be reporting e.g. if during any reconciliation more pods than expected were not ready, or sth like that]

Reply:
added some more data

Reply:
Sorry - let me be more explicit. This isn't actionable for cluster operators at all.

An operator cannot track upgrades of individual statefulsets and check what is happening during those.
The operator usually also cannot trigger actions on existing StatefulSets or created their own.

Metrics are the mechanism for this.

Reply:
changed to TODO for when we target beta. @kow3ns  if you have advice on this or adding something to this effect in alpha